repos/system_upgrade/el7toel8/actors/satelliteupgradefacts/actor.py

```diff
@@ -0,0 +1,133 @@
+import os
+import textwrap
+
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import run
+from leapp.models import InstalledRPM, Module, RepositoriesSetupTasks, RpmTransactionTasks
+from leapp.tags import FactsPhaseTag, IPUWorkflowTag
+
+
+class SatelliteUpgradeFacts(Actor):
+    """
+    No documentation has been provided for the satellite_upgrade_facts actor.
+    """
+
+    name = 'satellite_upgrade_facts'
+    consumes = (InstalledRPM, )
+    produces = (RepositoriesSetupTasks, RpmTransactionTasks, reporting.Report)
+    tags = (IPUWorkflowTag, FactsPhaseTag)
+
+    def process(self):
+        has_foreman = has_package(InstalledRPM, 'foreman') or has_package(InstalledRPM, 'foreman-proxy')
+        if not has_foreman:
+            return
+
+        local_postgresql = has_package(InstalledRPM, 'rh-postgresql12-postgresql-server')
+        postgresql_contrib = has_package(InstalledRPM, 'rh-postgresql12-postgresql-contrib')
+        postgresql_evr = has_package(InstalledRPM, 'rh-postgresql12-postgresql-evr')
+        # remove packages we don't need anymore
+        self.produce(RpmTransactionTasks(
+            to_remove=['tfm-runtime', 'tfm-pulpcore-runtime', 'rh-redis5-runtime', 'rh-ruby27-runtime', 'rh-python38-runtime'],
+            # you'll need to define the repos in leapp_upgrade_repositories.repo
+            # until this is on the CDN
+            to_install=['rubygem-foreman_maintain', 'foreman-installer-katello'],
```

Comment:
Isn't `foreman-installer-katello` here covered by lines 61 & 62 now?

Reply:
it is, yeah